frontend/src/app/containers/App/index.js

```diff
@@ -109,21 +105,6 @@ class App extends Component {
   }
 
   componentDidMount() {
-    const userAgent = navigator.userAgent.toLowerCase();
-    this.props.setHasAddon(!!window.navigator.testpilotAddon);
```

Comment:
I started to pull this thread... "Can I remove this line?" and wound up unraveling the rest of addon and browser initialization. It turns out `shouldShowUpgradeWarning` relies on a set of assumptions that came unglued when `hasAddon` doesn't start out as null. Anyway, the browser state is static once initialized, so I moved it into the initial state, which saves a lap around the event loop and also ensure the rest of the conditions for `shouldShowUpgradeWarning` are in sync.